ghost/link-tracking/lib/LinkClickTrackingService.js

```diff
@@ -78,7 +97,102 @@ class LinkClickTrackingService {
         });
     }
 
-    /** 
+    /**
+   * validate and manage the new redirect url in filter
+   * `to` url needs decoding and transformation to relative url for comparision
+   * @param {string} filter
+   * @returns {Object} parsed filter
+   * @throws {errors.BadRequestError}
+   */
+    #parseLinkFilter(filter) {
+        // decode filter to manage any encoded uri components
+        filter = decodeURIComponent(filter);
```

Comment:
If we do stick with using NQL filter - IMO the decoding of URI component should happen way closer to the HTTP level, in the endpoint/controller. Not a big deal but this service shouldn't need to be concerned with the encoding of data in transit

Reply:
🤔 I am ok to move this up to endpoint controller, I had originally thought about it as well but it felt inconsistent tbh to have bits split between controller and service as for other places, tho i didn't feel too strongly either ways. Gonna try moving this up in this case to controller, is it good if that happens in a separate change ?